google-cloud-pubsub/lib/google/cloud/pubsub/subscriber.rb

```diff
@@ -66,20 +69,28 @@ def stop
           synchronize do
             return if @request_queue.nil?
             @request_queue.push self
+            @inventory.clear
```

Comment:
Messages removed from inventory can still be processed. What we want is when the Subscriber stops to stop the Subscriber's background thread, _and_ the Inventory's background thread. so I'm using `Inventory#clear` as a way to stop the Inventory's background thread.

But, you bring up a good point, messages can still be processed by the Subscriber's thread pool and that will add the messages to Inventory. I suppose what we need is to add a `Inventory#stop` method that can stop the background thread. `Inventory#clear` is too much of a cheat. :)

I'll update the PR. Thanks for pointing this out!